### PR TITLE
Fix strlen deprecation in User

### PR DESCRIPTION
**Pull request by dpfaffenbauer**

### CODE REVIEW

File: models/User.php [hunk 1]
```diff
@@ -147,7 +147,7 @@ public function getPassword()
      */
     public function setPassword($password)
     {
-        if (strlen($password) > 4) {
+        if (!empty($password) && strlen($password) > 4) {
```
Comment (blankse) [inline models/User.php:150]:
```suggestion
        if (is_string($password) && strlen($password) > 4) {
```
What value was $password? null? Than we should also update the phpdoc.

Comment (dpfaffenbauer) [reply models/User.php:150]: What do you mean? $password is supposed to be a string, as the PHPDoc also says.

Comment (blankse) [reply models/User.php:150]: When it is always a string, than we should not need this check. It must be a other type.

Comment (brusch) [reply models/User.php:150]: @blankse is right, must have been something other than `string` - I'd assume it was `null`.

**Comment by brusch** [comment]
Closed in favor of #11792

**Pull request closed without merging**